src/components/CommentCard.jsx

```diff
@@ -1,10 +1,55 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../contexts/User";
+import { deleteCommentById } from "../api";
 
-export default function CommentCard({ comment }) {
+export default function CommentCard({ comment, setComments }) {
   const parsedDate = Date.parse(comment.created_at);
+
   const formattedDate = new Date(parsedDate).toLocaleString("en-GB", {
     timeZone: "UTC",
   });
+
+  const { loggedInUser, setLoggedInUser } = useContext(UserContext);
+  const [deleteBtn, setDeleteBtn] = useState(false);
+  const [deleteComment, setDeleteComment] = useState({});
+
+  useEffect(() => {
+    loggedInUser.username === comment.author
+      ? setDeleteBtn(true)
+      : setDeleteBtn(false);
+  }, [loggedInUser]);
+
+  useEffect(() => {
+    if (deleteComment.comment_id) {
+      setTimeout(
+        () =>
+          setComments((currCommets) => {
+            return [...currCommets];
+          }),
+        100
+      );
```

Comment:
This logic would want to be in the .then after the comment deletion, not in a useEffect. Also, you shouldn't really have to use setTimeout at all here either